### DISCUSSION TITLE
v1.5.0 Pre-release

### DISCUSSION THREAD

**Discussion by cvergne** [cvergne/vscode-php-getters-setters#19] [Announcements]
I've just published a v1.5.0 pre-release of the extension.
As the Github Action (_and so OVSX_) does not support pre-release, I've published it manually on VSCode Marketplace.

This version has the following changes :
- **Bumped minimal VSCode version to 1.63 to handle pre-release**
- Add a new option to generate or not PHPDoc, _enabled by default_
- Add a new option to enable PHP 7+ type hints, _enabled by default_
- Handle nullable types in setter param and getter return (*require PHP 7+ type hints option to be enabled*)
- Handle PHP 8 properties
- Handle properties with comments

It's a beta as there is no written test for the moment (_the original extension did not had one_), and there is several major changes.

So it should fix #11, #15, #5 and #10.

I don't know if the template feature is really used, I'll probably remove it in a 2.0 release (_leave a comment with why is it useful to you if you're using it_)

**Note:** VSCode 1.63+ is required to install pre-releases. As VSCode doesn't handle yet semver for pre-release, the final release should be published as 1.6.0 then.

**Comment by alcalyn** [comment]
> I don't know if the template feature is really used, I'll probably remove it in a 2.0 release

If you're talking about editing a custom `getter.js` and `setter.js`, I'm using it because the style of getter&setter is often different between projects
